homeassistant/components/remote_calendar/diagnostics.py

```diff
@@ -0,0 +1,23 @@
+"""Provides diagnostics for local calendar."""
+
+import datetime
+from typing import Any
+
+from homeassistant.core import HomeAssistant
+from homeassistant.util import dt as dt_util
+from homeassistant.const import CONF_URL
+from . import RemoteCalendarConfigEntry
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: RemoteCalendarConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator = entry.runtime_data
+    return {
+        "now": dt_util.now().isoformat(),
+        "timezone": str(dt_util.get_default_time_zone()),
+        "system_timezone": str(datetime.datetime.now().astimezone().tzinfo),
+        "url": entry.data[CONF_URL],
+        "data": repr(coordinator.data),
```

Comment:
what kind of object is this? a dataclass? Use `asdict` instead then

Reply:
It inherits from a pydantic BaseModel.
asdict is not working:
`  File "/workspaces/core/homeassistant/components/remote_calendar/diagnostics.py", line 22, in async_get_config_entry_diagnostics
    "data": asdict(coordinator.data),
            ~~~~~~^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/dataclasses.py", line 1358, in asdict
    raise TypeError("asdict() should be called on dataclass instances")
TypeError: asdict() should be called on dataclass instances
`
But just `dict`?

Reply:
Oh I believe that pydantic has its own way to serialize to a dict. I would recommend doing this, while this also does give the info, it's annoying to read and work with

Reply:
Maybe it's better to use the original input from the remote, like in the local calendar integration. I changed it to this.

Reply:
Hold on, this is the entire users calendar! You can't put that in diagnostics without redaction. Yes, follow what `local_calendar` is doing since its the same object type.